contrib/platform/src/com/sun/jna/platform/win32/WinNT.java

```diff
@@ -3126,33 +3134,89 @@ public void read() {
+         * processors belong to one NUMA node.
          */
         public int nodeNumber;
 
         /**
          * This member is reserved.
          */
-        public byte[] reserved = new byte[20];
+        public byte[] reserved = new byte[18];
+
+        /**
+         * The number of groups included in the GroupMasks array. This field was
+         * introduced in TBD Release Iron. On earlier versions, this value is always 0.
+         */
+        public short groupCount;
 
         /**
          * A {@link GROUP_AFFINITY} structure that specifies a group number and
-         * processor affinity within the group.
+         * processor affinity within the group. This member is only relevant if
+         * {@code groupCount} is 0.
          */
         public GROUP_AFFINITY groupMask;
 
+        /**
+         * An array of {@link GROUP_AFFINITY} structures that specifies a group number
+         * and processor affinity within the group. This member is only relevant if
+         * {@code groupCount} is 1 or greater.
+         */
+        public GROUP_AFFINITY[] groupMasks = new GROUP_AFFINITY[1];
+
         public NUMA_NODE_RELATIONSHIP() {
         }
 
         public NUMA_NODE_RELATIONSHIP(Pointer memory) {
             super(memory);
         }
+
+        /*
+         * Since the groupMasks array always has at least one element we copy it to the
+         * groupMask field for backwards compatibility in the case groupCount is 0 or 1.
+         * Otherwise we resize the array.
+         */
+        @Override
+        public void read() {
+            readField("groupCount");
+            if (groupCount > 1) {
+                groupMasks = new GROUP_AFFINITY[groupCount];
+            }
+            super.read();
+            // Copy first value from array to older version of structure
+            groupMask = groupMasks[0];
+        }
+
+        /*
+         * getFieldList and getFieldOrder are overridden because there are two versions
+         * of this structure. We will always populate the groupMasks field in the newer
+         * version, but we will copy groupMasks[0] to the groupMask field elsewhere for
+         * backwards compatibility.
+         */
+        @Override
+        protected List<Field> getFieldList() {
+            List<Field> fields = new ArrayList<Field>(super.getFieldList());
+            Iterator<Field> fieldIterator = fields.iterator();
+            while (fieldIterator.hasNext()) {
+                Field field = fieldIterator.next();
+                if ("groupMask".equals(field.getName())) {
+                    fieldIterator.remove();
+                }
+            }
+            return fields;
+        }
+
+        @Override
+        protected List<String> getFieldOrder() {
+            List<String> fieldOrder = new ArrayList<String>(super.getFieldOrder());
+            fieldOrder.remove("groupMask");
+            return fieldOrder;
```

Comment:
Am I missing something obvious? Why don't you just remove `groupMask` from the field list and get rid of the `getFieldList` and `getFieldOrder` overrides?

Reply:
Huh.  I was under the impression that the list of `public` fields had to match up exactly or we'd get the "does not provide enough" version of this error, as the field list is populated via reflection on public instance variables.

https://github.com/java-native-access/jna/blob/5ac61611f3c4a74ca3c6ed05c4ffafd7c6070e30/src/com/sun/jna/Structure.java#L1101-L1111

But that doesn't seem to be the case.

Reply:
```
import com.sun.jna.Structure;
import com.sun.jna.Structure.FieldOrder;

public class Test {

    @FieldOrder({ "baf" })
    public static class Foo extends Structure {
        public int bar;
        public int baz;
    }

    public static void main(String[] args) {
        Foo foo = new Foo();
        System.out.println(foo.size());
    }
}
```
Result:
```
Exception in thread "main" java.lang.Error: Structure.getFieldOrder() on class oshi.Test$Foo does not provide enough names [1] ([baf]) to match declared fields [2] ([bar, baz])
	at com.sun.jna.Structure.getFields(Structure.java:1110)
	at com.sun.jna.Structure.deriveLayout(Structure.java:1265)
	at com.sun.jna.Structure.calculateSize(Structure.java:1192)
	at com.sun.jna.Structure.allocateMemory(Structure.java:413)
	at com.sun.jna.Structure.ensureAllocated(Structure.java:389)
	at com.sun.jna.Structure.ensureAllocated(Structure.java:379)
	at com.sun.jna.Structure.size(Structure.java:446)
	at oshi.Test.main(Test.java:16)
```

Reply:
So I can remove it from the `@FieldOrder` and remove the `getFieldOrder()` override but I still need the `getFieldList()` override.

Reply:
It wasn't quite that obvious -- in my initial testing without reading size I didn't get the error and got really confused.  And you were 2/3 right.

I had a passing thought about an optional second argument to `@FieldOrder` that allowed you to specify the same list as the field list.   But the use cases are probably rare enough that's not needed.